src/vcpkg/base/system.mac.cpp

```diff
@@ -0,0 +1,244 @@
+    {
+        static constexpr char capital_hexits[] = "0123456789ABCDEF";
+        // 6 hex bytes, 5 dashes
+        static constexpr size_t MAC_STRING_LENGTH = MAC_BYTES_LENGTH * 2 + 5;
+
+        char mac_address[MAC_STRING_LENGTH];
+        char* mac = mac_address;
+        unsigned char c = static_cast<unsigned char>(bytes[0]);
+        *mac++ = capital_hexits[(c & 0xf0) >> 4];
+        *mac++ = capital_hexits[(c & 0x0f)];
+        unsigned char non_zero_mac = c;
+        for (size_t i = 1; i < MAC_BYTES_LENGTH; ++i)
+        {
+            c = static_cast<unsigned char>(bytes[i]);
+            *mac++ = '-';
+            *mac++ = capital_hexits[(c & 0xf0) >> 4];
+            *mac++ = capital_hexits[(c & 0x0f)];
+            non_zero_mac |= c;
+        }
+
+        return non_zero_mac ? std::string(mac_address, MAC_STRING_LENGTH) : "";
+    }
+
+    std::string preferred_interface_or_default(const std::map<StringView, std::string>& ifname_mac_map)
+    {
+        // search for preferred interfaces
+        for (auto&& interface_name : {"eth0", "wlan0", "en0", "hn0"})
+        {
+            auto maybe_preferred = ifname_mac_map.find(interface_name);
+            if (maybe_preferred != ifname_mac_map.end())
+            {
+                return maybe_preferred->second;
+            }
+        }
+
+        // default to first mac address we find
+        auto first = ifname_mac_map.begin();
+        if (first != ifname_mac_map.end())
+        {
+            return first->second;
+        }
+
+        return "0";
+    }
+}
+#endif
+
+namespace vcpkg
+{
+    std::string get_user_mac_hash()
+    {
+#if defined(_WIN32)
+        // getmac /V /NH /FO CSV
+        // outputs each interface on its own comma-separated line
+        // "connection name","network adapter","physical address","transport name"
+        static constexpr StringLiteral ZERO_MAC = "00-00-00-00-00-00";
+        static constexpr size_t CONNECTION_NAME = 0;
+        static constexpr size_t PHYSICAL_ADDRESS = 2;
+        auto getmac = cmd_execute_and_capture_output(
+            Command("getmac").string_arg("/V").string_arg("/NH").string_arg("/FO").string_arg("CSV"));
+        if (getmac.exit_code != 0)
+        {
+            return "0";
+        }
+
+        std::map<std::string, std::string> ifname_mac_map;
+        for (auto&& line : Strings::split(getmac.output, '\n'))
+        {
+            auto values = Strings::split(line, ',');
+            if (values.size() != 4)
+            {
+                return "0";
+            }
+
+            auto&& name = Strings::replace_all(values[CONNECTION_NAME], "\"", "");
+            auto&& mac = Strings::replace_all(values[PHYSICAL_ADDRESS], "\"", "");
+            if (mac == ZERO_MAC) continue;
+            ifname_mac_map.emplace(name, Hash::get_string_hash(mac, Hash::Algorithm::Sha256));
+        }
+
+        // search for preferred interfaces
+        for (auto&& interface_name : {"Local Area Connection", "Ethernet", "Wi-Fi"})
```

Comment:
Aren't these names localized?

Reply:
Yes, pretty sure they are. I think returning first non-zero should be OK